src/dsp_tools/commands/xmlupload/xmlupload.py

```diff
@@ -407,6 +411,8 @@ def _upload_resources(
 
         res = None
         try:
+            if i >= interrupt_after:
+                raise KeyboardInterrupt(f"Interrupted: Maximum number of resources was reached ({interrupt_after})")
```

Comment:
EDIT: i think these lines should be at the very beginning of the for loop, i.e. on line 405